### DISCUSSION TITLE
DKIM case sensitive domain sign problem

### DISCUSSION THREAD

**Discussion by nurjns** [rspamd/rspamd#4698] [Ideas]
Hello,

We recognized that mails are not signed by DKIM when the user is using capital letters of the domain in his mail software, while the domain in the configuration consists only of lowercase letters, e.g.: test@DomaiN.de instead of test@domain.de

Rspamd apparently interprets this as case sensitive and sees it as a different domain, for which there is of course no key in the configuration.

Is it possible to make a fix for this?

**Comment by fatalbanana** [comment]
It sounds like a bug but I can't reproduce it, usually that stuff gets lower-cased.

**Reply by nurjns** [reply]
Please check this thread: https://community.keyhelp.de/viewtopic.php?t=12816&sid=1c7acdeba2ac0026ce9ec29443ebd892

**Reply by fatalbanana** [reply]
So it's specific to `signing_table`.

Reason seems to be that the `glob` map type which is used here is case-sensitive.

**Reply by vstakhov** [reply]
I think it is simple just to lowercase domain name when doing a lookup.

**Reply by fatalbanana** [reply]
Yes indeed, we can do that.

**Reply by fatalbanana** [reply]
https://github.com/rspamd/rspamd/pull/4717

**Reply by nurjns** [reply]
I see that this should be fixed, but I still get this message with mail-tester:

"DKIM_SIGNED: Message has a DKIM or DK signature, not necessarily valid"

If I send in lower case letters everything is OK.

**Reply by fatalbanana** [reply]
I see no likely reason for it, do you run 3.7.5 indeed?

**Reply by fatalbanana** [reply]
I'm not familiar with what mail-tester should output but this sounds like it received a signed message.

**Reply by nurjns** [reply]
Yes, my panel "KeyHelp" is using 3.7.5 now. I tested it by sending from the shell with the mail-command, so the mails should not be signed. And if I just change the sender address to loser case letters all is fine.

**Reply by nurjns** [reply]
It looks it it's only a "problem" by the workflow of mail-tester.com. DKIM_SIGNED is negative, but for this DKIM_VALID is positive. And result of other mail-testers say DKIM is fine.
